### PR TITLE
[FUN FRIDAY] Use vulture to detect unused code

### PR DESCRIPTION
**Pull request by r0x0d**
Added vulture to pre-commit to detect unused code in our codebase.

Jira Issues: [RHELC-](https://issues.redhat.com/browse/RHELC-)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [ ] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: scripts/whitelist.py
Comment (SpyTec) [inline scripts/whitelist.py:1]: What's the point of this file?

File: .pre-commit-config.yaml
Comment (SpyTec) [inline .pre-commit-config.yaml:1]: Can we get vulture as one commit and the fixes in another?

File: scripts/whitelist.py
Comment (r0x0d) [reply scripts/whitelist.py:1]: This file is supposed to whitelist every "unused code" that we can't really remove right now. 

It is a bypass in vulture to not detect those lines as unused.

File: .pre-commit-config.yaml
Comment (r0x0d) [reply .pre-commit-config.yaml:1]: done.

File: scripts/whitelist.py
Comment (r0x0d) [reply scripts/whitelist.py:1]: Added a comment to the top of the level explaining briefly what this file does.

File: scripts/get_copr_build_id.py
Comment (SpyTec) [inline scripts/get_copr_build_id.py:1]: I wonder, should we exclude `scripts/*` completely?

**Review by SpyTec — COMMENTED** [review]
Overall looks good! Just think we should exclude the scripts dir as it won't get called normally

**Review by SpyTec — APPROVED** [review]

**Pull request merged**